app/src/main/java/com/example/rickandmortyapp/helper/SavedStateFlow.kt

```diff
@@ -0,0 +1,88 @@
+package com.example.rickandmortyapp.helper
```

Comment:
Do not try to understand this file, this is useful to better manage the ViewState being stored inside the SavedState handle.

Consider this: the ViewModel outlives the Fragment lifecycle, but the ViewModel can still be killed, so we want a way to always enable the ViewModel to recover the data that was there before. 

The Saved State is the way to persist such data. So when we initialize the ViewState inside the ViewModel, we do this through the SavedStateHandle, and this is the code that controls such logic.